src/main.js

```diff
@@ -1594,12 +1597,18 @@ function populateDiscoveryFeedModelFromQueries(model, proxies, recommended) {
     }).catch(e => logError(e, 'Query failed'));
 }
 
+const AUTO_CLOSE_MILLISECONDS_TIMEOUT = 12000;
+
 const DiscoveryFeedApplication = new Lang.Class({
     Name: 'DiscoveryFeedApplication',
     Extends: Gtk.Application,
 
     _init: function() {
-        this.parent({ application_id: pkg.name });
+        this.parent({
+            application_id: pkg.name,
+            inactivity_timeout: AUTO_CLOSE_MILLISECONDS_TIMEOUT,
+            flags: Gio.ApplicationFlags.IS_SERVICE
```

Comment:
Setting this flag will actually break using the "debug window" feature; you either have to manually call `activate()` on the application in that case, or use `--gapplication-service`. Since you prefer the explicit flag, I pushed a commit that does that.